cpp/CMakeLists.txt

```diff
@@ -18,6 +18,8 @@
 cmake_minimum_required(VERSION 2.7)
 project(arrow)
 
+set(ARROW_VERSION "0.2.0")
```

Comment:
I would prefer if we could read them from some generated location. In Python we use at the moment `git describe` and as a fallback the Java `pom.xml` to determine the version number as those two get automatically updated on a release.